site/content/en/docs/Getting started/first_kustomization.md

```diff
@@ -57,10 +107,251 @@ spec:
         app: nginx
     spec:
       containers:
-      - name: nginx
-        image: nginx:latest
+      - image: nginx
+        name: nginx
         ports:
         - containerPort: 80
+```
+
+### What are Bases and Overlays?
+
+Let's think of a scenario where we need to deploy the nginx manifests from the previous section to 2 environments called `Staging` and `Production`, however although most of the manifest is the same for both environments we need to apply some environment specific changes.  Traditionally we could duplicate the manifests and apply the changes manually or rely on some templating engine. With kustomize we can avoid templating and duplication of our manifests and apply the different changes we need using overlays.  With this approach, the `base` would contain the common part of the our manifests and the `overlays` contain our environment specific changes.
+
+To continue with this tutorial, create the `kustomization.yaml` files for our two overlays:
+
+```bash
+cat <<'EOF' >overlays/staging/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../base
 EOF
+
+cat <<'EOF' >overlays/production/kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ../../base
+EOF
+```
+
+The `kustomization.yaml` files in the overlays are just including the `base` folder, so if you were to run `kustomize build` on the overlay folders at this point you would get the same output we got when we built `base`. The directory structure you created so far should look like this:
+
+```
+kustomize-example
+├── base
+│   ├── deployment.yaml
+│   ├── kustomization.yaml
+│   └── service.yaml
+└── overlays
+    ├── production
+    │   └── kustomization.yaml
+    └── staging
+        └── kustomization.yaml
+```
+
+### Customising our overlays
+
+For the purposes our example, let's define some requirements of how our deployment should look like in the two environments:
+
+|Requirement|         Production         |         Staging         |
+|-----------|----------------------------|-------------------------|
+|Name       |overlay1-nginx-production   |overlay2-nginx-staging   |
+|Namespace  |production                  |staging                  |
+|Replicas   |3                           |2                        |
+
+
+We can achieve the names required by making use of `namePrefix` and `nameSuffix` as follows:
+
+
+_Production overlay kustomization.yaml_:
```

Comment:
I like the suggestion, have updated this as well.  It would be nice if there could be a way to make this comment be an actual title for the code block, but haven't figured out if there's a way to do it or if it is possible yet